### PR TITLE
Add default_systcls option to crio.conf

### PR DESCRIPTION
**Pull request by umohnani8**
Users can add default sysctls that they want all cri-o containers
to have in the crio.conf file now.
Adding net.ipv4.ping_group_range as a default so we can drop the
NET_RAW capability.

Remove sysctl parsing code from cri-o as we extract the sysctls from k8s directly.
Had a TODO from a while back to remove it, so removing.

Drop the NET_RAW capability from the list of cri-o defaults
as we are using the sysctl net.ipv4.ping_group_range instead
to allow users to do a ping.

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan PTAL. For now the default sysctl is `net.ipv4.ping_group_range=0 0`, what should I change the value of this sysctl to be?

**Comment by mrunalp** [comment]
@umohnani8 You will have to restrict it to /proc/1/gid_map values on the extreme else sysctl will fail.

File: lib/config.go [hunk 1]
```diff
@@ -67,6 +66,11 @@ var DefaultCapabilities = []string{
 	"KILL",
 }
 
+// DefaultSysctls for the default-sysctls option in the crio.conf file
+var DefaultSysctls = []string{
+	"net.ipv4.ping_group_range=0 0",
```
Comment (mrunalp) [inline lib/config.go]: We cannot hard code this. For user namespaces, the gid_map could be different so will have to populate the range and validate it against user ns group mappings.

Comment (rhatdan) [reply lib/config.go]: I guess we need to see if these are relative to the USERNS.  IE Does this happen in runc after the USERNS is setup, then it should work.

Comment (giuseppe) [reply lib/config.go]: it should happen from inside the container once the user namespace is configured.  Some sysctl are not allowed from a namespace and the OCI runtime will fail to set these, but I don't think this PR has to worry of it.

**Comment by umohnani8** [comment]
@mrunalp the tests that ping a pod from another pod should be testing this change automatically - https://github.com/kubernetes-incubator/cri-o/blob/master/test/network.bats#L103.

**Comment by mrunalp** [comment]
We can skip the test for RHEL 7 by checking for the setting on the host.

**Comment by mrunalp** [comment]
Just debugged this with @umohnani8. This sysctl isn't available when using a user namespace on RHEL. We should use some other sysctl for testing such as ` net.ipv4.ip_forward`.

**Comment by umohnani8** [comment]
Tests are green! @mrunalp PTAL

File: server/sandbox_run_linux.go [hunk 2]
```diff
@@ -406,7 +407,21 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		g.AddAnnotation(k, v)
 	}
 
+	// Add default sysctls given in crio.conf
+	for _, defaultSysctl := range s.config.DefaultSysctls {
+		split := strings.Split(defaultSysctl, "=")
```
Comment (rhatdan) [inline server/sandbox_run_linux.go]: Should this be a SplitN(defaultSysctl, "=",2)
TO make sure the sysctl does not include a "=", I know it is doubtful, but...

File: server/sandbox_run_linux.go [hunk 3]
```diff
@@ -406,7 +407,21 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
...
+		if len(split) == 2 {
+			if err := validateSysctl(split[0], hostNetwork, hostIPC); err != nil {
+				logrus.Warnf("%v - skipping...", err)
```
Comment (rhatdan) [inline server/sandbox_run_linux.go]: skipping %s ..., err, defaultSysctl
Should this be an error and refuse to start the container?

Comment (umohnani8) [reply server/sandbox_run_linux.go]: I was thinking why error out completely when it is just possible that the user made a typo or something in the sysctls they have enabled. Made more sense to log it and just continue to create the error.
If you think it is better to fail completely I can change it.

Comment (rhatdan) [reply server/sandbox_run_linux.go]: Well we at lease need more info on which sysctl failed.

Comment (umohnani8) [reply server/sandbox_run_linux.go]: Oh, yeah adding that.

File: server/sandbox_run_linux.go (hunk 2 above)
Comment (umohnani8) [reply server/sandbox_run_linux.go]: Fixed.

File: server/sandbox_run_linux.go (hunk 3 above)
Comment (umohnani8) [reply server/sandbox_run_linux.go]: Fixed.

**Comment by umohnani8** [comment]
@mrunalp this is ready to merge :)

**Pull request merged**